### DISCUSSION TITLE
Docker hub image -- How to start?

### DISCUSSION THREAD

**Discussion by 4jag** [inventree/InvenTree#2740] [General]
This page [Docker Image](https://inventree.readthedocs.io/en/latest/start/docker/) refers to a [page with a pull command](https://hub.docker.com/r/inventree/inventree) -- what can be done with that?

EDIT:

Sorry, you're right--there's no mention of `docker pull` on the first page. But it does say:
"The most convenient method of installing and running InvenTree is to use the official [docker image](https://hub.docker.com/r/inventree/inventree), available from docker-hub"

Re your comment below (cuz I'm banned) ...

Exactly! That's what I'm trying to do -- I want to stand up an as-is instance to show it to our production group. Then I'll get the build process sorted out so we make customizations.  So I've been trying to get this (production) image running locally as a demo. The trouble I'm having is that I'm not sure how to combine the pre-built image with configuration, since the config yaml files are supposed to be _inside_ the image when it starts.

BTW, I've moved into WSL2/Ubuntu for docker-build since it seems others posting here have succeeded, but still stuck on the below where alpine is not reachable. The error says to check the repositories file, so I don't know if it DNS or I need to update alpine's location
 
`WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.13/main: could not connect to server (check repositories file)
WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.13/community: could not connect to server (check repositories file)
ERROR: unable to select packages:
  bash (no such package):
    required by: world[bash]
  cairo (no such package):
    required by: world[cairo]
  cairo-dev (no such package):
    required by: world[cairo-dev]
.... (~40-50 more)
`

**Comment by matmair** [comment]
Hi there

> This page [Docker Image](https://inventree.readthedocs.io/en/latest/start/docker/) refers to a [docker pull command](https://hub.docker.com/r/inventree/inventree) -- what can be done with that?

This question is factually false.
The first link leads to the documentation on how the image needs to be set up (which [environment variables](https://inventree.readthedocs.io/en/latest/start/docker/#environment-variables) must be configured, that [a volume should be mounted](https://inventree.readthedocs.io/en/latest/start/docker/#data-directory) and that the code ships with sample [setup files](https://inventree.readthedocs.io/en/latest/start/docker/#data-directory) to get the image running).  
The second link leads to an external container registry. That registry automatically includes a reference to a pull command - which is not mentioned once in the whole documentation of Inventree as the image needs to be provided with variables before you get a running container.

If you want to learn more about how we use docker and what differences exist between production and development setups I suggest reading [our introduction to development setups](https://inventree.readthedocs.io/en/latest/start/docker_dev/#docker-development-server).
The (production) image is more catered to users who are already familiar with the documentation and inner workings of the docker setup. It is often used for docker, Kubernetes or docker swarm deployments with orchestrators (like Rancher or Portainer) in production in companies that use InvenTree as their PLM.
